File: java/src/main/java/api/jdy/FileDemo.java
package api.jdy;

import constants.HttpConstant;
import model.http.HttpRequestParam;
import org.apache.commons.lang3.StringUtils;
import util.HttpUtil;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

import static constants.HttpConstant.APP_BASE_URL;

public class FileDemo {

    /**
     * 获取文件上传凭证和上传地址接口
     *
     * @param appId
     * @param entryId
     * @param transactionId
     * @return
     * @throws Exception
     */
    public Map<String, Object> uploadToken(String appId, String entryId, String transactionId) throws Exception {
        if (StringUtils.isBlank(appId) || StringUtils.isBlank(entryId) || StringUtils.isBlank(transactionId)) {
            throw new RuntimeException("param lack!");
        }
        String url = APP_BASE_URL + appId + "/entry/" + entryId + "/file/get_upload_token";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        // 请求参数
        Map<String, Object> data = new HashMap<>();
        data.put("transaction_id", transactionId);
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

    public Map<String, Object> uploadFile(String url, String token, File file) throws Exception {
        if (StringUtils.isBlank(url) || StringUtils.isBlank(token) || file == null) {
            throw new RuntimeException("param lack!");
        }
        return HttpUtil.httpPostFile(url, token, file);
    }

}

File: python/src/api/jdy/file_demo.py
from ...constants.http_constant import HttpConstant
from ...model.http.http_request_param import HttpRequestParam
from ...model.http.http_request_param import generateTransactionId
from ...util.http_util import ApiClient

# 合法的版本
valid_versions = ('v1')
# 默认版本
default_version = 'v1'


class FileApiClient(ApiClient):

    def __init__(self, api_key, host):
        ApiClient.__init__(self, api_key, host, valid_versions, default_version)

    """
    获取文件上传凭证和上传地址接口
    Arguments:
        work_flow_query_param: WorkFlowApprovalCommentQueryParam 实例
    """

    def uploadToken(self, app_id, entry_id, version=default_version):
        url = HttpConstant.FILE_UPLOAD_URL.format(app_id=app_id, entry_id=entry_id,
                                                  version=self.getValidVersion(version))
        request_param = HttpRequestParam(HttpConstant.API_KEY, url, generateTransactionId())
        return self.send_post(request_param)

    """
    上传文件
    Arguments:
        url: str 
        token: str
        file: 
    """

    def uploadFile(self, url, token, file, version=default_version):
        data = {'token': token}
        return self.send_post_with_file(url, data, file)
